### ISSUE TITLE
Problem with vanilla HTTP server

### ISSUE THREAD

**Issue by Asuza** [caolan/dispatch#6]
Despite having a very simple server set up, I receive a script error when the dispatcher is called.
```
/path/node_modules/dispatch/lib/dispatch.js:77
            next();
            ^
TypeError: undefined is not a function
    at Server.<anonymous> (/path/node_modules/dispatch/lib/dispatch.js:77:13)
    at Server.emit (events.js:98:17)
    at HTTPParser.parser.onIncoming (http.js:2108:12)
    at HTTPParser.parserOnHeadersComplete [as onHeadersComplete] (http.js:121:23)
    at Socket.socket.ondata (http.js:1966:22)
    at TCP.onread (net.js:527:27)
```

Here is the script I'm using:
```javascript
var os = require('os'),
    http = require('http'),
    dispatch = require('dispatch'),
    server,
    port;

server = http.createServer(
    dispatch({
        '/': function (request, response) {
            response.end();
        }
    })
);

port = process.argv[2];

console.log('Listening for requests at http://' + os.hostname() + ':' + port);
server.listen(port);
```

I've triggered the error several times simply by visiting the web address of the server. Do you know what's wrong?

**Comment by emanuelpalm** [comment]
I've fixed the error with #7. I hope to get merged soon.

**Comment by samizzo** [comment]
I'm seeing this too. It seems like it was never tested with node's http server since the `next` argument is specific to the `connect` module. To fix it it's enough to just check that `next` is non-null before calling it.

**Comment by Asuza** [comment]
@emanuelpalm, it's been two months now... are you able to merge it?

**Comment by emanuelpalm** [comment]
@Asuza Unfortunately, I don't have any particular repo privileges.

**Comment by Asuza** [comment]
@emanuelpalm, I apologize. I mistakenly thought you were the repo owner.

**Comment by Asuza** [comment]
I believe this issue is fixed thanks to #7.
